lib/secure_headers/headers/content_security_policy.rb

```diff
@@ -212,5 +212,9 @@ def strip_source_schemes(source_list)
     def symbol_to_hyphen_case(sym)
       sym.to_s.tr("_", "-")
     end
+
+    def source_scheme(source)
+      URI(source.sub(PORT_WILDCARD_REGEX, '')).scheme
```

Comment:
Unfortunately, this calculates the source scheme of `example.org:443` as `example.org`

Reply:
This might work well enough for us.
```suggestion
      source.split("://")[-2] || source.split(":")[-2]
```

Reply:
I feel safer still using URI parsing. I've amended `source_scheme` so that it considers this special case where the actual scheme is missing.